### PR TITLE
add systemd-udev

### PR DESCRIPTION
**Pull request by Pixelnarium**
Since udev-175 is broken right now and probably will also break in the future I think it is a good idea to offer an alternative to the udev sources from 2012.
I saw that [the flatpaked Steam](https://github.com/flathub/com.valvesoftware.Steam) pulls udev from systemd which is maintained. I tried it with my packages and it seems to work quite well (within the boundaries of flatpak of course).

I don't know the rules of shared-modules but I still ask myself if the filename itself should be versioned. udev-175 is... other shared-modules are not (i.e. glew). systemd updates come quite often but I think the udev code is not changed that much but that is just a guess. I think it should not be versioned to reduce the maintenance burden of packages who are using it. But in this PR it is versioned to be in line with udev-175. The version also comes with the shared-modules submodule commit reference automatically. New versions are not shoved into every packages automatically and untested.

Maybe also asking @nanonyme and @LeandroStanger (who are working on this in the Steam Flatpak) what they think about this. Maybe I missed something here.

### CODE REVIEW

**Comment by LeandroStanger** [comment]
@ramcq @gasinvein This will be useful for Lutris Flatpak

**Comment by LeandroStanger** [comment]
@RobLoach @gasinvein This will be useful for Lutris Flatpak

File: udev/systemd-udev-241.json [hunk 1]
```diff
@@ -0,0 +1,79 @@
+{
+  "name": "hwdata",
+  "config-opts": [
+    "--datarootdir=/app/share"
+  ],
+  "sources": [
+    {
+      "url": "https://github.com/vcrhonek/hwdata/archive/v0.321.tar.gz",
```
Comment (LeandroStanger) [inline udev/systemd-udev-241.json]: Update to hwdata v0.321

File: udev/systemd-udev-241.json [hunk 2]
```diff
@@ -0,0 +1,79 @@
+{
+  "name": "hwdata",
```
Comment (gasinvein) [inline udev/systemd-udev-241.json]: hwdata not needed for systemd.

Comment (Pixelnarium) [reply udev/systemd-udev-241.json]: So it is not part of the systemd-hwdb stuff? It looked like something udev uses. Sure, will remove if it is totally unrelated and unneeded.

**Comment by Erick555** [comment]
Maybe [this](https://github.com/gentoo/eudev) would be better alternative for bringing udev which doesn't need to compile whole systemd?

**Comment by eszlari** [comment]
> Maybe this would be better alternative for bringing udev which doesn't need to compile whole systemd?

But most parts can be disabled in meson. And I'm sure upstream (being mostly Redhat people) would be open to add more options (if needed) to support the flatpak use case.

**Comment by Pixelnarium** [comment]
> Maybe [this](https://github.com/gentoo/eudev) would be better alternative for bringing udev which doesn't need to compile whole systemd?

As far as I can see the end result is pretty much just the udev library even if it compiles a little bit more, it is deleted afterwards. Same with eudev since it also a little more than just libudev.so. I think in the end it is just a matter of taste and I have no personal preference on that. I just took the approach from the Steam flatpak to start the discussion. If more people want to use eudev it is fine by me.

**Comment by Erick555** [comment]
Flatpak usecase is to make udev as dumb as possible as most of it's functionality won't work anyway therefore I think eudev is better suited for. the job.

> As far as I know flatpak also includes some parts of systemd. There is a libsystemd.so in the runtime.

It was added to be able to run runtime as standalone system in vm. For flatpak it's useless.

**Comment by eszlari** [comment]
> I think eudev is better suited for the job.

Depends on have well that fork is maintained. I think it makes more sense to work on something like
```
meson -Dudev_only_build=true
```
 for systemd than rely on some fork.

File: udev/systemd-udev-241.json (hunk 2 above)
Comment (gasinvein) [reply udev/systemd-udev-241.json]: usbutils and pciutils uses it. Systemd has it's own [copy](/systemd/systemd/tree/master/hwdb) of `pci.ids` and `usb.ids.

Comment (Pixelnarium) [reply udev/systemd-udev-241.json]: Okay. I removed it.

**Comment by Pixelnarium** [comment]
It was already in there the whole time. Just thought that hwdata is part of this.

**Comment by gasinvein** [comment]
@Pixelnarium  Oh, sorry, didn't notice it.

**Comment by Erick555** [comment]
> Depends on have well that fork is maintained. I think it makes more sense to work on something like

it's critical part of most non-systemd distros so it have to be maintained well 😄 

> meson -Dudev_only_build=true

Then why it's not used here?

**Comment by eszlari** [comment]
> Then why it's not used here?

I meant that such a option could be added to systemd's meson build system.

**Comment by gasinvein** [comment]
Is this still WIP?

**Comment by Pixelnarium** [comment]
The filename ist pretty much the last question (except from updating systemd itself).

**Comment by Pixelnarium** [comment]
That's my offer for now.

**Comment by TingPing** [comment]
This seems fine, if a bit much, but I'd like to see this tested by some current udev module users in practice to ensure everything is good.

**Comment by gasinvein** [comment]
@TingPing it's currently used by Steam, see flathub/com.valvesoftware.Steam#178.

**Comment by Erick555** [comment]
@gasinvein I think that an app used old udev should be tested against the new one.

**Comment by Lctrs** [comment]
I tested [GitKraken](https://github.com/flathub/com.axosoft.GitKraken) with a custom local build of the [Electron2 base app](https://github.com/flathub/org.electronjs.Electron2.BaseApp) including systemd-udev this morning and it worked well.

**Comment by Erick555** [comment]
@Lctrs speaking of GitKraken, you may be interested [in this fixes for git build](https://github.com/flathub/org.qownnotes.QOwnNotes/pull/9).

**Comment by jjardon** [comment]
FYI, freedesktop-sdk 19.08 will include libudev (from systemd)

**Comment by Erick555** [comment]
@TingPing freedesktop-sdk 19.08 was released with libudev. This can be closed.

**Pull request closed without merging**